awx/main/tasks.py

```diff
@@ -396,47 +406,74 @@ def purge_old_stdout_files():
             logger.debug("Removing {}".format(os.path.join(settings.JOBOUTPUT_ROOT, f)))
 
 
+@task(queue=get_local_queuename)
+def check_heartbeat(node):
+    AWXReceptorJob.check_heartbeat(node)
+
+
+def discover_receptor_nodes():
+    ctl = ReceptorControl(RECEPTOR_SOCK)
+    connections = ctl.simple_command('status')['Advertisements']
```

Comment:
The per-controller-node AWX heartbeat task *now* speaks to its local receptor and attempts to discover and register *new* Instances.  When a new instance is registered, its capacity is initially *zero*, but the first thing we do is call `check_heartbeat.apply_async([hostname])`, which dispatches an adhoc command to that node to calculate (and report) its runner version and capacity.